cmd/operator/main.go

```diff
@@ -108,12 +112,21 @@ func main() {
 		os.Exit(1)
 	}
 
+	client := mgr.GetClient()
+	operatorDeployment := &appsv1.Deployment{}
+	if err = client.Get(context.TODO(), types.NamespacedName{Name: operatorDeploymentName, Namespace: ns}, operatorDeployment); err != nil {
+		setupLog.Error(err, "Unable to get Infrastructure Operator Deployment")
+		os.Exit(1)
+	}
```

Comment:
I was considering whether this belongs to the `Reconcile` function or not. I think it's fine here since there's no way to change these values without restarting the `infrastructure-operator` pod, which will restart the reconcile and, therefore, the assisted-service deployment.

Reply:
I was thinking about it for quite some time too :-)